reactnative-todo/src/hooks/journey-hook.js

```diff
@@ -0,0 +1,224 @@
+/*
+ * forgerock-sample-web-react
+ *
+ * state.js
+ *
+ * Copyright (c) 2021 ForgeRock. All rights reserved.
+ * This software may be modified and distributed under the terms
+ * of the MIT license. See the LICENSE file for details.
+ */
+import { useContext, useEffect, useState } from 'react';
+import { NativeModules } from 'react-native';
+import { FRStep } from '@forgerock/javascript-sdk';
+import { useNavigation } from '@react-navigation/native';
+import { AppContext } from '../global-state';
+
+const { FRAuthBridge } = NativeModules;
+/**
+ *
+ * @param {Object} props - React props object
+ * @param {Object} props.action - Action object for a "reducer" pattern
+ * @param {string} props.action.type - Action type string that represents the action
+ * @param {Object} props.form - The form metadata object
+ * @returns {Object} - React component object
+ */
+function useJourneyHandler({ action }) {
+  /**
+   * Compose the state used in this view.
+   * First, we will use the global state methods found in the App Context.
+   * Then, we will create local state to manage the login journey. The
+   * underscore is an unused variable, since we don't need the current global state.
+   *
+   * The destructing of the hook's array results in index 0 having the state value,
+   * and index 1 having the "setter" method to set new state values.
+   */
+  // Form level errors
+  const [formFailureMessage, setFormFailureMessage] = useState(null);
+  // Step to render
+  const [renderStep, setRenderStep] = useState(null);
+  // Step to submit
+  const [submissionStep, setSubmissionStep] = useState(null);
+  // Processing submission
+  const [submittingForm, setSubmittingForm] = useState(false);
+  // User state
+  const [user, setUser] = useState(null);
+  const [, { setAuthentication }] = useContext(AppContext);
+  const navigation = useNavigation();
+
+  /**
+   * Since we have API calls to AM, we need to handle these requests as side-effects.
+   * This will allow the view to render, but update/re-render after the request completes.
+   */
+  useEffect(() => {
+    /**
+     * @function getOAuth - The function to call when we get a LoginSuccess
+     * @returns {undefined}
+     */
+    async function getOAuth() {
+      /** *********************************************************************
+       * SDK INTEGRATION POINT
```

Comment:
Let's call out "NATIVE BRIDGE SDK" here, rather than just SDK. This gives people the ability to do quick project wide text searches for what's used where.